services/web/server/src/simcore_service_webserver/projects/_db_utils.py

```diff
@@ -335,28 +336,55 @@ async def _get_project(
         # this retrieves the projects where user is owner
         user_groups: list[RowProxy] = await self._list_user_groups(connection, user_id)
 
-        # NOTE: ChatGPT helped in producing this entry
-        conditions = sa.and_(
-            projects.c.uuid == f"{project_uuid}",
-            projects.c.type == f"{ProjectType.TEMPLATE.value}"
-            if only_templates
-            else True,
-            sa.or_(
-                projects.c.prj_owner == user_id,
-                sa.text(
-                    f"jsonb_exists_any(projects.access_rights, {assemble_array_groups(user_groups)})"
-                ),
-            ),
+        # Helper subquery that prepares access rights data in
+        # backwards compatible json type.
+        access_rights_subquery = (
```

Comment:
A note and a request at the same time.

I'm not sure why this has to be written in a way that it is retro compatible with the old SQL.

After the entire CI is green, would it not make more sense to drop the old "json" format and use a join for computing access rights? I think, that in the long run this will make the logic more simple.

Reply:
Sorry maybe the comment was misleading I will remove it. This is a optimal way how you can get access rights from the DB table in json format. I just wanted to point out that I did it in a way that it will not destroy any interface. Also the new `project_to_groups` table is already used there utilizing DB engine power! :)